### ISSUE TITLE
Memory leaks messages when app undeployed in Tomcat

### ISSUE THREAD

**Issue by tunix** [hazelcast/hazelcast#7831]
I have a Spring Boot 1.3.2 app that is deployed into Tomcat 8 (working with Java 8) as WAR file. Tomcat is working inside a Docker container. There is nothing special about the configuration; it's basically a fork of tutum's tomcat images.

When I undeploy my app, I get the following messages in Tomcat's log:

```
2016-03-24 14:45:15.126  INFO 1 --- [nio-8080-exec-5] o.s.c.support.DefaultLifecycleProcessor  : Stopping beans in phase 0
2016-03-24 14:45:15.160  INFO 1 --- [nio-8080-exec-5] o.s.j.e.a.AnnotationMBeanExporter        : Unregistering JMX-exposed beans on shutdown
2016-03-24 14:45:15.170 DEBUG 1 --- [nio-8080-exec-5] c.m.myapp.conf.HazelcastConfig       : Shutting down Hazelcast instance _hzInstance_1_dev..
24-Mar-2016 14:45:15.240 SEVERE [http-nio-8080-exec-5] org.apache.catalina.loader.WebappClassLoaderBase.checkThreadLocalMapForLeaks The web application [MyApp] created a ThreadLocal with key of type [com.hazelcast.util.UuidUtil$1] (value [com.hazelcast.util.UuidUtil$1@34d8f6af]) and a value of type [java.util.Random] (value [java.util.Random@596914c]) but failed to remove it when the web application was stopped. Threads are going to be renewed over time to try and avoid a probable memory leak.
24-Mar-2016 14:45:15.244 SEVERE [http-nio-8080-exec-5] org.apache.catalina.loader.WebappClassLoaderBase.checkThreadLocalMapForLeaks The web application [MyApp] created a ThreadLocal with key of type [com.hazelcast.spi.impl.operationservice.impl.BackpressureRegulator$1] (value [com.hazelcast.spi.impl.operationservice.impl.BackpressureRegulator$1@6d2d7a37]) and a value of type [java.util.Random] (value [java.util.Random@1745cf58]) but failed to remove it when the web application was stopped. Threads are going to be renewed over time to try and avoid a probable memory leak.
24-Mar-2016 14:45:15.796 INFO [http-nio-8080-exec-5] org.apache.catalina.startup.HostConfig.undeploy Undeploying context [/MyApp]
```

* Hazelcast Version: 3.5.5
* Number of clients: 1
* Java Version: Java 8

**Comment by tunix** [comment]
Closing this duplicate ticket in favor of #4354 .

**Comment by Donnerbart** [comment]
This has been fixed in Hazelcast 3.8 via https://github.com/hazelcast/hazelcast/pull/9691
